### PR TITLE
Add accessibilityRole and accessibilityLevel props

### LINKED ISSUE #635: Support for h1-6 tags on react-native-web
Allow `accessibilityRole` prop on Text, and add `aria-level` prop. 

More info:
https://github.com/necolas/react-native-web/commit/1b172319b9ea257830a82cb6800ad3e7a746190c

### PR DESCRIPTION
**Pull request by Naturalclar**
Closes #635 

Added `accessibilityRole` and `accessibilityLevel` props to Text component.

accessibilityLevel prop does not exist in `react-native` but will work in `react-native-web`

Ref: https://github.com/necolas/react-native-web/pull/402/files

### CODE REVIEW

**Comment by jfrolich** [comment]
Wow that was fast!

**Comment by Naturalclar** [comment]
@jfrolich I noticed after I sent the PR that the actual props for `accessibilityLevel` is supposed to be `aria-level` 
I'm trying to find out whether I can set prop names with hyphen as a prop.

**Comment by Naturalclar** [comment]
If I'm understanding [this](https://reasonml.github.io/reason-react/docs/en/invalid-prop-name) correctly,
ariaLevel will be translated into `aria-level` under `reason-react`
So maybe, setting an ariaLevel props will get the job done?

**Comment by jfrolich** [comment]
Perhaps in the same PR we can also add the extra accessibility roles that are valid in `react-native-web`:

https://github.com/necolas/react-native-web/blob/36dacb2052efdab2a28655773dc76934157d9134/packages/react-native-web/src/modules/AccessibilityUtil/propsToAccessibilityComponent.js#L13-L23.

File: src/components/Text.re [hunk 1]
```diff
@@ -8,6 +8,47 @@ external make:
     ~accessible: bool=?,
     ~accessibilityHint: string=?,
     ~accessibilityLabel: string=?,
+    ~accessibilityRole: [@bs.string] [
+                          | // below are additional roles added for react-native-web https://github.com/necolas/react-native-web/blob/36dacb2052efdab2a28655773dc76934157d9134/packages/react-native-web/src/modules/AccessibilityUtil/propsToAccessibilityComponent.js#L13-L23
```
Comment (MoOx) [inline src/components/Text.re]: You can see that the compiler is moving the comment up there (which sucks but that's a fact...)
What about instead keeping everything Alpha ordered with just `// web` after web only values?

Comment (jfrolich) [reply src/components/Text.re]: Or a general comment for the accessibility roles enumerating all the variants that apply to react-native-web. This is probably also better for tools that extract documentation from the source.

Comment (Naturalclar) [reply src/components/Text.re]: @MoOx @jfrolich 
Thanks for the suggestion! :)
I've listed the strings in alphabetical order, and added comment about react-native-web specific roles.

**Pull request merged**